src/main/java/com/microsoft/sqlserver/jdbc/Parameter.java

```diff
@@ -445,14 +397,10 @@ boolean isValueGotten() {
 
     Object getValue(JDBCType jdbcType, InputStreamGetterArgs getterArgs, Calendar cal, TDSReader tdsReader,
             SQLServerStatement statement) throws SQLServerException {
-        if (null == getterDTV) {
+        if (null == getterDTV)
             getterDTV = new DTV();
-        }
-
-        if (null != tdsReader) {
-            deriveTypeInfo(tdsReader);
-        }
 
+        deriveTypeInfo(tdsReader);
```

Comment:
why remove the null check?

Reply:
TdsReader should not be null. Before the revert, it could potentially be null because we may or may not have the value type info depending on how the parameters were set. eg. either set by name or by index for the cstmt.